scripts/run-ku.sh

```diff
@@ -13,11 +13,14 @@ KU_TMP_DIR="$2"
 
 # Abort if the device is currently plugged in, as that's liable to confuse Nickel into actually starting a real USBMS session!
 # Which'd probably ultimately cause a crash with our shenanigans...
-if [ $(cat /sys/devices/platform/pmic_battery.1/power_supply/mc13892_charger/online) -ne 0 ]; then
-    # Sleep a bit to lose the race with Nickel's opening of our image
-    sleep 2
-    logmsg "C" "Device is currently plugged in. Aborting!"
-    exit 1
+# Except if we've specified 'allowUSBPower = true' in our ku.toml file
+if ! grep -qi "allowUSBPower *= *true" "/mnt/onboard/${KU_DIR}/config/ku.toml"; then 
```

Comment:
I'd strip the spaces for the globbing to match if someone were to edit the config without spaces (`allowUSBPower=false`).

i.e., `grep -qi 'allowUSBPower*=*true'`

Reply:
Um, that string is a regex. Your proposed modification would fail quite spectacularly if there are any spaces.

Reply:
Perhaps I should have used the more verbose form ` {0,}` instead of ` *` to avoid confusion

Reply:
Um, maybe read that command again ... remember grep searches by regexes, not globs. With what you suggested, it would match `allowUSBPowerr===true` and `allowUSBPowetrue`, but not `allowUSBPower = true`. :smiley:

Reply:
Just switched to using the `[[:blank:]]` character class instead. Should also catch tabs, if anyone uses those instead of spaces.

Reply:
Huh. I was somehow convinced that grep -F was the default behavior. Mind blown :D.

Reply:
Well, there was an oopsie in the regex after all :p 

It would have quite happily matched if the string was inside a comment, or the option line was commented out.

The regex now matches [line start][zero or more whitespace][option]

Reply:
Well, one option might be to have a config query command in ku. Then, you'd be able to use the usual parser for it, and it wouldn't be much more overhead than grep.

Reply:
If I end up having more options, that might be an something I'll look into.

A bit overkill at this stage though.